modules/tests/src/test/scala/dev/profunktor/redis4cats/TestScenarios.scala

```diff
@@ -575,4 +578,52 @@ trait TestScenarios { self: FunSuite =>
       _ <- IO(assert(c4 > 0, "merged hyperloglog should think it has more than 0 items in"))
     } yield ()
   }
+
+  def keyPatternSubScenario(client: RedisClient): IO[Unit] = {
+    import dev.profunktor.redis4cats.effect.Log.NoOp._
+
+    val pattern = "__keyevent*__:*"
+    val key     = "somekey"
+    val resources = for {
+      commands <- Redis[IO].fromClient(client, RedisCodec.Utf8)
+      sub <- PubSub.mkSubscriberConnection[IO, String, String](client, RedisCodec.Utf8)
+      stream <- Resource.pure(sub.psubscribe(RedisPattern(pattern)))
+      listener <- Resource.eval(stream.head.compile.toList.start)
+      _ <- Resource.eval(commands.set(key, ""))
+      _ <- Resource.eval(commands.expire(key, 1.seconds))
+      firstEvent <- Resource.eval(
+                     listener.joinWith(IO.raiseError(new IllegalStateException("Fiber should not be cancelled")))
+                   )
+    } yield firstEvent.headOption
+    resources.use(IO.pure).map { result =>
```

Comment:
Can't this be just `resources.use { result => ... }`? Same in the other scenario.